File: gameFramework/src/main/java/at/grueneis/game/framework/code/AndroidSound.java
package at.grueneis.game.framework.code;

import android.media.SoundPool;
import at.grueneis.game.framework.Sound;

public class AndroidSound implements Sound
{
	int soundID;
	SoundPool soundPool;
	
	public AndroidSound(int soundID, SoundPool soundPool)
	{
		this.soundID = soundID;
		this.soundPool = soundPool;
	}
	
	public void play(float volume)
	{
		soundPool.play(soundID, volume, volume, 0, 0, 1);
	}
	
	public void dispose()
	{
		soundPool.unload(soundID);
	}
	
}

File: gameFramework/src/main/java/at/grueneis/game/framework/code/AndroidSound.kt
package at.grueneis.game.framework.code

import android.media.SoundPool
import at.grueneis.game.framework.Sound

class AndroidSound(var soundID: Int, var soundPool: SoundPool) : Sound {
    override fun play(volume: Float) {
        soundPool.play(soundID, volume, volume, 0, 0, 1f)
    }

    override fun dispose() {
        soundPool.unload(soundID)
    }
}